src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -1486,6 +1486,17 @@ ntp_archiver::maybe_upload_aborted_tx(
     co_return std::nullopt;
 }
 
+ss::future<> ntp_archiver::upload_index(
+  ss::sstring path, cloud_storage::offset_index index) {
+    retry_chain_node rtc{
+      _conf->segment_upload_timeout(),
```

Comment:
this might be a bit too generous... could pass in the remaining budget from the rtc node at the call site or something like that

Reply:
That should be fine, we don't have to align index uploads with segment uploads in time. They could happen way later in the background.